js/src/instructions/item.ts

```
@@ -0,0 +1,1227 @@
+          )[0],
+          receiver: (this.program.client.provider as AnchorProvider).wallet
+            .publicKey,
+          systemProgram: SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+        })
+        .remainingAccounts(remainingAccounts)
+        .instruction(),
+    ];
+  }
+
+  async beginItemActivation(
+    args: BeginItemActivationArgs,
+    accounts: BeginItemActivationAccounts,
+    _additionalArgs: BeginItemActivationAdditionalArgs = {}
+  ) {
+    const remainingAccounts =
+      await generateRemainingAccountsGivenPermissivenessToUse({
+        permissivenessToUse: args.usagePermissivenessToUse,
+        tokenMint: accounts.itemMint,
+        parentMint: args.itemClassMint,
+        parentIndex: args.classIndex,
+        parent: (await getItemPDA(args.itemClassMint, args.classIndex))[0],
+        metadataUpdateAuthority: accounts.metadataUpdateAuthority,
+        program: this.program.client,
+      });
+
+    const itemClassKey = (
+      await getItemPDA(args.itemClassMint, args.classIndex)
+    )[0];
+
+    const [itemActivationMarker, itemActivationBump] =
+      await getItemActivationMarker({
+        itemMint: accounts.itemMint,
+        index: args.index,
+        usageIndex: new BN(args.usageIndex),
+        amount: args.amount,
+      });
+
+    const instructions: TransactionInstruction[] = [];
+    const itemTransferAuthority = accounts.itemTransferAuthority;
+
+    if (
+      accounts.itemAccount &&
+      accounts.itemAccount.equals(
+        (
+          await getAtaForMint(
+            accounts.itemMint,
+            (this.program.client.provider as AnchorProvider).wallet.publicKey
+          )
+        )[0]
+      )
+    ) {
+      instructions.push(
+        Token.createApproveInstruction(
+          TOKEN_PROGRAM_ID,
+          accounts.itemAccount,
+          itemTransferAuthority.publicKey, // QUESTION: itemTransferAuthority could be null - what to do then?
```

If `accounts.itemAccount` exists, the if statement is truthy,  and `itemTransferAuthority` is null, we should throw an exception since we won't be able to transfer the item to the escrow when activating it.